scripts/api-inference/scripts/generate.ts

```diff
@@ -99,10 +99,43 @@ const TASKS_DATA = (await response.json()) as any;
 //// Snippet utils ////
 ///////////////////////
 
+const formatSnippets = (result: snippets.types.InferenceSnippet | snippets.types.InferenceSnippet[], defaultClient: string, language: string): string => {
+  // For single snippet, return just the content
+  if (!Array.isArray(result) || result.length === 1) {
+    const snippet = Array.isArray(result) ? result[0] : result;
+    return snippet.content;
+  }
+  
+  // Get the appropriate comment 
+  const commentPrefix = {
+    'py': '#',
+    'js': '//',
+    'bash': '#'
+  }[language] || '#';
+  
+  // Show the snippet for each client
+  return result
+    .map(snippet => 
+      `${commentPrefix} With ${snippet.client || defaultClient} client\n\n${snippet.content}`
+    )
+    .join('\n\n');
+};
```

Comment:
(nit) I'd make each client appear in a separate snippet if possible. What's nice with a "self-contained" snippets is that you can copy them in one click. If both clients are in the same code snippet, you can't really do that anymore
 
![image](https://github.com/user-attachments/assets/899bb079-dc03-4732-877d-b419cfc3dc48)

Reply:
addressed in [7cfb641](https://github.com/huggingface/hub-docs/pull/1490/commits/7cfb64175913af2470e2c5f547e295e98f907f7a) but the PR doc does not seem to be updated yet

Reply:
nevermind, the doc is now updated 
![Screenshot 2024-11-15 at 16 12 30](https://github.com/user-attachments/assets/dd1ae951-90cd-4bca-b3ff-efbedfc60224)